### PR TITLE
StreamPayment status Runtime API and RPC

### PR DESCRIPTION
**Pull request by nanocryk**
Adds a Runtime API and RPC to fetch the status of a stream payment, which consists of the pending payment at a given moment in time, and if the payment is stalled or not. It's goal is to be used in the data providers implementation to follow assignments and their payment, and stop providing the service if the stream payment is stalled (which otherwise "occurs" passively without any extrinsic or events). 

TODO:
- [x] Figure out why the new test fails (request acceptation doesn't seem to succeed while it succeed in a similar other test)

### CODE REVIEW

File: test/suites/dev-tanssi/stream-payment/test_stream_payment_rpc.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+                expect(requestChangeEvents.length).to.be.equal(1);
+
+                expect(await rpcStreamPaymentStatus(context, newBlock.block.hash, 0, null)).to.deep.equal({
+                    deposit_left: 9_800_000,
+                    stalled: true,
+                    payment: 0,
+                });
+
+                let stream = await polkadotJs.query.streamPayment.streams(0);
+                console.log(stream.toString());
```
Comment (girazoki) [inline test/suites/dev-tanssi/stream-payment/test_stream_payment_rpc.ts:123]:
```suggestion
```

File: test/suites/dev-tanssi/stream-payment/test_stream_payment_rpc.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+                expect(await rpcStreamPaymentStatus(context, newBlock.block.hash, 0, null)).to.deep.equal({
+                    deposit_left: 9_800_000,
+                    stalled: true,
+                    payment: 0,
+                });
+
+                // produce empty block on session change, which cannot contain extrinsics
+                await context.createBlock();
+
+                // 5th block: accept change, resuming stream
```
Comment (girazoki) [inline test/suites/dev-tanssi/stream-payment/test_stream_payment_rpc.ts:137]:
```suggestion
                // 6th block: accept change, resuming stream
```

File: test/suites/dev-tanssi/stream-payment/test_stream_payment_rpc.ts [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+                });
+                expect(acceptChangeEvents.length).to.be.equal(1);
+
+                expect(await rpcStreamPaymentStatus(context, newBlock.block.hash, 0, null)).to.deep.equal({
+                    deposit_left: 9_805_000, // old deposit + increase
+                    stalled: false,
+                    payment: 0,
+                });
+
+                // 6th block: create an empty block to check status
```
Comment (girazoki) [inline test/suites/dev-tanssi/stream-payment/test_stream_payment_rpc.ts:154]:
```suggestion
                // 7th block: create an empty block to check status
```

**Review by girazoki — APPROVED** [review]
LGTM overall!

File: pallets/stream-payment/rpc/runtime-api/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Tanssi.  If not, see <http://www.gnu.org/licenses/>
+
+//! Runtime API for Stream Payment pallet
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+extern crate alloc;
```
Comment (tmpolaczyk) [inline pallets/stream-payment/rpc/runtime-api/src/lib.rs:21]:
```suggestion
```

Comment (nanocryk) [reply pallets/stream-payment/rpc/runtime-api/src/lib.rs:21]: How do you import `String` in `no_std` without `alloc`?

Comment (tmpolaczyk) [reply pallets/stream-payment/rpc/runtime-api/src/lib.rs:21]: Sorry, I though `use alloc` already imported the crate. Maybe we can import `String` from `sp_std` instead?

**Review by tmpolaczyk — APPROVED** [review]

**Pull request merged**